src/bin/pg_autoctl/cli_config.c

```diff
@@ -632,6 +632,157 @@ cli_config_set(int argc, char **argv)
 }
 
 
+static bool
+cli_keeper_config_validate_replication(KeeperConfig *config, PGSQL *pgsql)
+{
+	Keeper keeper = { 0 };
+	NodeAddress primary = { 0 };
+	bool primaryUnreachable;
+
+	if (!keeper_init(&keeper, config))
+	{
+		/* already logged */
+		return false;
+	}
+
+	/* get the primary info to populate the replication source */
+	if (!monitor_get_primary(&keeper.monitor,
+							 config->formation,
+							 keeper.state.current_group,
+							 &primary))
+	{
+		log_warn("Monitor could not be contacted for primary conninfo, skipping "
+				 "password validation");
+		return true;
+	}
+
+	if (primary.nodeId == keeper.state.current_node_id)
```

Comment:
It's a little strange to decide if we're a primary or not with that check, but given that we need the primary information for the connection string later, why not. I guess I'm saying that a comment about this choice would make sense.